src/analysis/retail/mage/frost/talents/GlacialSpike.tsx

```diff
@@ -18,92 +20,52 @@ class GlacialSpike extends Analyzer {
-
-    this.addEventListener(
-      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.GLACIAL_SPIKE_TALENT),
-      this.onGlacialSpikeCast,
-    );
-    this.addEventListener(
-      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.GLACIAL_SPIKE_DAMAGE),
-      this.onGlacialSpikeDamage,
-    );
-    this.addEventListener(Events.fightend, this.onFightEnd);
   }
 
-  onGlacialSpikeCast(event: CastEvent) {
-    if (this.lastCastEvent) {
-      this.flagTimeline(this.lastCastEvent);
-    }
+  nonShatterCasts = () => {
+    const spikeCasts = this.eventHistory.getEvents(EventType.Cast, {
+      spell: TALENTS.GLACIAL_SPIKE_TALENT,
+    });
 
-    this.lastCastEvent = event;
-    this.lastCastDidDamage = false;
-  }
-
-  onGlacialSpikeDamage(event: DamageEvent) {
-    if (!this.lastCastEvent) {
-      return;
-    }
-    if (!HasTarget(this.lastCastEvent)) {
-      return;
-    }
-
-    const castTarget = encodeTargetString(
-      this.lastCastEvent.targetID,
-      this.lastCastEvent.targetInstance,
+    //If Icy Veins was active, disregard it
+    let badCasts = spikeCasts.filter(
+      (gs) => !this.selectedCombatant.hasBuff(TALENTS.ICY_VEINS_TALENT.id, gs.timestamp),
     );
-    const damageTarget = encodeTargetString(event.targetID, event.targetInstance);
-
-    //We dont care about the Glacial Spikes that split to something else via Splitting Ice.
-    if (castTarget !== damageTarget) {
-      return;
-    }
 
-    this.lastCastDidDamage = true;
-    const enemy: any = this.enemies.getEntity(event);
-    if (enemy && SHATTER_DEBUFFS.some((effect) => enemy.hasBuff(effect.id, event.timestamp))) {
-      this.spikeShattered += 1;
-    } else {
-      this.spikeNotShattered += 1;
-      this.flagTimeline(this.lastCastEvent);
-    }
-    this.lastCastEvent = undefined;
-  }
-
-  onFightEnd(event: FightEndEvent) {
-    if (this.lastCastEvent) {
-      this.flagTimeline(this.lastCastEvent);
-    }
-  }
-
-  flagTimeline(event: CastEvent) {
-    if (!this.lastCastEvent) {
-      return;
-    }
-
-    event.meta = event.meta || {};
-    event.meta.isInefficientCast = true;
-    if (this.lastCastDidDamage) {
-      event.meta.inefficientCastReason = `You cast Glacial Spike without shattering it. You should wait until it is frozen or you are able to use a Brain Freeze proc to maximize its damage.`;
-    } else {
-      event.meta.inefficientCastReason =
-        'The target died before Glacial Spike hit it. You should avoid this by casting faster spells on very low-health targets, it is important to not waste potential Glacial Spike damage.';
-    }
-  }
+    //If Glacial Spike was shattered on the main target, disregard it
+    badCasts = badCasts.filter((gs) => {
+      const spikeDamage = this.eventHistory.getEvents(EventType.Damage, {
+        searchBackwards: false,
+        spell: SPELLS.GLACIAL_SPIKE_DAMAGE,
+        count: 1,
+        startTimestamp: gs.timestamp,
+        targetID: gs.targetID,
```

Comment:
this should also pass `targetInstance` or if you glacial spike something that has multiple instances, you'll find a bunch of possibly-unrelated matches